cmd/unpack/main.go

```diff
@@ -42,35 +45,69 @@ func main() {
 			}
 
 			bundleFS := os.DirFS(bundleDir)
-			bundleContents := map[string][]byte{}
+			buf := &bytes.Buffer{}
+			gzw := gzip.NewWriter(buf)
+			tw := tar.NewWriter(gzw)
 			if err := fs.WalkDir(bundleFS, ".", func(path string, d fs.DirEntry, err error) error {
 				if err != nil {
 					return err
 				}
+
+				if d.Type()&os.ModeSymlink != 0 {
+					return nil
+				}
 				if bundleDir == "/" {
 					// If bundleDir is the filesystem root, skip some known unrelated directories
 					fullPath := filepath.Join(bundleDir, path)
 					if skipRootPaths.Has(fullPath) {
 						return filepath.SkipDir
 					}
 				}
-				if d.IsDir() {
-					return nil
+				info, err := d.Info()
+				if err != nil {
+					return fmt.Errorf("get file info for %q: %w", path, err)
 				}
-				data, err := fs.ReadFile(bundleFS, path)
+
+				h, err := tar.FileInfoHeader(info, "")
 				if err != nil {
-					return fmt.Errorf("read file %q: %w", path, err)
+					return fmt.Errorf("build tar file info header for %q: %w", path, err)
+				}
+				h.Uid = 0
+				h.Gid = 0
+				h.Uname = ""
+				h.Gname = ""
+				h.Name = path
+
+				if err := tw.WriteHeader(h); err != nil {
+					return fmt.Errorf("write tar header for %q: %w", path, err)
+				}
+				if !d.IsDir() {
+					data, err := fs.ReadFile(bundleFS, path)
+					if err != nil {
+						return fmt.Errorf("read file %q: %w", path, err)
+					}
+					if _, err := tw.Write(data); err != nil {
+						return fmt.Errorf("write tar data for %q: %w", path, err)
+					}
 				}
-				bundleContents[path] = data
 				return nil
 			}); err != nil {
-				log.Fatalf("walk bundle dir %q: %v", bundleDir, err)
+				log.Fatalf("generate tar.gz for bundle dir %q: %v", bundleDir, err)
 			}
-
-			encoder := json.NewEncoder(os.Stdout)
-			if err := encoder.Encode(bundleContents); err != nil {
+			if err := tw.Close(); err != nil {
+				log.Fatal(err)
+			}
+			if err := gzw.Close(); err != nil {
 				log.Fatal(err)
 			}
+
+			bundleMap := map[string]interface{}{
+				"content": buf.Bytes(),
+			}
+			enc := json.NewEncoder(os.Stdout)
```

Comment:
This is probably a poor idea: should we add a flag that allows us to configure the file descriptor that's used to relay unpacked Bundle contents? Right now, we always default to stdout (e.g. the Pod logs) but I wonder if we'll ever need to write directly to a file?

Reply:
Also: can we use io.Copy here?

Reply:
> should we add a flag that allows us to configure the file descriptor that's used to relay unpacked Bundle contents?

I figure we can add a flag for that if/when we need it?

> can we use io.Copy here?

Not sure I'm following this suggestion (though I did see that I could use `io.Copy` in the WalkDir function to write the file contents from a file handle rather than reading the file into memory and then writing that `[]byte` back out).

Reply:
Yep sgtm. Nothing blocking here imo.